test/cached_csi_test.go

```diff
@@ -187,3 +231,26 @@ func TestCachedCSIDriverProbeFailsUntilPrepared(t *testing.T) {
 func formatFileMode(mode os.FileMode) string {
 	return fmt.Sprintf("%#o", mode)
 }
+
+func createTestCachedServer(tc util.TestCtx, tmpDir string) (*api.Cached, string, func()) {
+	cl, _, closeClient := createTestClient(tc)
+	_, grpcServer, _ := createTestGRPCServer(tc)
+
+	s := cached.CachedServer{
+		Grpc: grpcServer,
+	}
+
+	cached := tc.CachedApi(cl, path.Join(tmpDir, "cached", "staging"))
+	s.RegisterCached(cached)
+	s.RegisterCSI(cached)
+
+	socket := path.Join(tmpDir, "csi.sock")
+	endpoint := "unix://" + socket
+
+	go func() {
+		err := s.Serve(endpoint)
+		require.NoError(tc.T(), err, "CSI Server exited")
+	}()
+
+	return cached, endpoint, func() { closeClient(); s.Grpc.Stop() }
+}
```

Comment:
Worth adding a test that proves that the inner DL client can rebuild using the overlay-powered cache presented to it? I am mostly just thinkin there may be other sneaky overlayfs options we need to add

Reply:
Sure, I can add that.

Reply:
Added above